Cleanup database image

znmeb opened the pull request:
This pull request cleans out all the extraneous packages from DOCKERFILE.db.development. It also closes issue #17 . I've tested it locally with `bin/build.sh -l` and `bin/start.sh -l`.

bhgrant8 commented:
Could you make this other small update, and i can close my other PR, make it a bit easier?

https://github.com/hackoregon/backend-examplar-2018/pull/19/commits/0c7df40aa2797bf0cb0062b1913b28c1f50c9172#diff-d6cd3dc57892dc83bf4994d16439d142R32

znmeb commented:
I think this is right - GitHub is reading my mind.

znmeb commented:
I'm currently testing the update to stretch - branch is `update-to-stretch`.

bhgrant8 commented:
I cannot not comment on question 1

On 2: This mapping is just for when you are connecting to the database container from your host directly, within the container network you will still use the 5432. I think the main conflict being if you are already running postgres on the default port on the host and want to connect to the containerized db as well?

3. I think it is best to keep this PR to questions on the DB container. feel free to open issue on the requirements.

znmeb commented:
@MikeTheCanuck 

1. The `dbsuper` user was a convenience I added for people using the PostGIS container from the command line. I can probably remove it without impacting the API usage - let me test that out.
2. I also added Linux users / database superusers for the teams, mirroring the accounts you have on the AWS instance. They need to be in the database for the API but I'm not sure they need to be Linux users or database superusers. This container is only supposed to talk to the API; as long as a backup can get restored to the right owner, I think the rest is YAGNI.

znmeb commented:
OK - latest commit cleans up the database users:
1. There is now only one database superuser - `postgres`.
2. The team database users have been added with the same permissions they have in AWS. They have their own password, common to all five. It's in `env.sample` as `DEVELOPMENT_TEAM_PASSWORD`.

P.S.: There are only three Grateful Dead songs? ;-)

znmeb commented:
Let's do one last review - I have the changes to the API Dockerfiles ready to go so I'm not blocked on this one.
